maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/booterclient/JarManifestForkConfiguration.java

```diff
@@ -116,10 +116,10 @@ private File createJar( @Nonnull List<String> classPath, @Nonnull String startCl
             StringBuilder cp = new StringBuilder();
             for ( Iterator<String> it = classPath.iterator(); it.hasNext(); )
             {
-                File classPathElement = new File( it.next() );
+                String classPathElement = it.next();
                 String uri = toClasspathElementUri( parent, classPathElement, dumpLogDirectory );
                 cp.append( uri );
-                if ( classPathElement.isDirectory() && !uri.endsWith( "/" ) )
+                if ( new File( classPathElement ).isDirectory() && !uri.endsWith( "/" ) )
```

Comment:
Files.isDirectory will be more consistent with the new use of Paths.
What do you think?

Reply:
@eolivelli 
yes, and moreover we should change the method signature from this:
`toClasspathElementUri( String parent, String classPathElement, File dumpLogDirectory )`
to this:
`toClasspathElementUri( Path parent, Path classPathElement, Path dumpLogDirectory )`
Which requires calling `dumpLogDir.toPath()` in `ForkStarter` (Line 580) but pls do not go deeper changing File to Path in all project since it is not a matter of this PR.

Reply:
It would be nice to have `Path parent` created once before the loop.

Reply:
In the scope of this change which is about using Paths it seemed to me strange to add a line which is using the legacy API.
Btw the result will be correct

Reply:
I am totally with you. Let's wait for @jglick until he makes all changes.
off-topic: We have one JUnit5 issue and some easy issues to fix. I guess we can release a new release candidate next week,

Reply:
> we should change the method signature

No, I deliberately used `String` parameters so as to make it possible to write unit tests. `Path`s can (as far as I know) only be constructed using platform-specific methods, making the tests behave differently according to the machine running them.

Reply:
> `Files.isDirectory` will be more consistent with the new use of `Path`s.

I chose to make a more minimal change to solve the immediate regression, rather than rewriting the whole class to use NIO methods.

Reply:
Thst's correct. The Strings are already platform specific.